runelite-client/src/main/java/net/runelite/client/plugins/grandexchangeaverage/GrandExchangeAverageOfferWindow.java

```diff
@@ -0,0 +1,94 @@
+package net.runelite.client.plugins.grandexchangeaverage;
+
+import com.google.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+
+@Slf4j
+class GrandExchangeAverageOfferWindow
+{
+
+
+	private final Client client;
+
+	@Inject
+	GrandExchangeAverageOfferWindow(Client client)
+	{
+		this.client = client;
+	}
+
+
+	public boolean isOpen()
+	{
+		//Find the Offer Container.
+		Widget widgetGrandExchangeOfferContainer = client.getWidget(WidgetInfo.GRAND_EXCHANGE_OFFER_CONTAINER);
+
+		//Return false if not opened.
+		if (widgetGrandExchangeOfferContainer == null ||
+				widgetGrandExchangeOfferContainer.isHidden())
+		{
+			return false;
+		}
+
+		//Get the first child which contains the title text.
+		Widget[] children = widgetGrandExchangeOfferContainer.getDynamicChildren();
+		Widget widgetGrandExchangeOfferContainerTitle = children[1];
+
+		return widgetGrandExchangeOfferContainerTitle.getText().contains("Set up offer");
+	}
+
+	public int getItemId()
+	{
+		//Widget IDs in Grand Exchange interface has the same ID for some reason.
+		Widget widgetGrandExchangeOfferElements = client.getWidget(WidgetInfo.GRAND_EXCHANGE_OFFER_TITLE);
+
+		if (widgetGrandExchangeOfferElements == null ||
+				widgetGrandExchangeOfferElements.isHidden())
+		{
+			return -1;
+		}
+
+		//The title we want is 21th child.
+		Widget[] children = widgetGrandExchangeOfferElements.getDynamicChildren();
+
+		//If itemid is -1 or 6512 (both invalid)
+		if (children[21].getItemId() == -1 || children[21].getItemId() == 6512)
```

Comment:
Just a personal preference, but I would declare `21`, `-1` and `6512` as variables, just to give better clarity of what they mean, for example
```
class GrandExchangeAverageOfferWindow
{
    private final Client client;
    private final int GRAND_EXCHANGE_TITLE = 21;
    private final int INVALID_ITEM_ID_1 = -1;
    private final int INVALID_ITEM_ID_2 = 6512;
    ...
```

_(You may come up with better variable names for them, but hopefully paints the right picture)_, that way you can then remove the comment as it'll be inferred